src/redux/features/plantings/plantingsSelectors.ts

```diff
@@ -43,11 +44,11 @@ export const selectPlantingsDateRange = (state: RootState, dateRange: string[],
   }) ?? [];
 
 export const selectPlantingProgressSubzones = createSelector(
-  [(state: RootState) => selectPlantingSites(state), (state: RootState) => selectPlantings(state)],
+  [(state: RootState) => selectSearchPlantingSites(state), (state: RootState) => selectPlantings(state)],
   (plantingSites, plantings) => {
-    if (plantings) {
+    if (plantingSites && plantings) {
       const plantingsBySubzone = getTotalPlantsBySubzone(plantings);
-      const totalPlantsBySite = getTotalPlantsBySite(plantings);
+      const totalPlantsBySite = getTotalPlantsBySite(plantingSites);
```

Comment:
Noticed on line 56 we are traversing all zones and subzones, this generates a full list of all subzones (slow and incorrect).
The correct result should only have plantings progress in reported.
If you get a chance, can you 
1. return only subzones that are in the plantings data?
2. instead of iterating over all zones and subzones, we could create a map of plantingSite->reportedPlants simply from the plantings data.. and here just use that information
example. iterate over plantings once and populate a map `plantingsProgressMap` of plantingSite -> plantings
here we just set `reported: plantingsProgressMap[plantingSite.id] ?? []`
something like that...

Reply:
Ok np, we can revisit later maybe. I already added a check to only populate those entries that have a totalSeedlingsSent - that takes care of one of my requests. 
Previously it was returning a list of all the subzones even if they didn't have plantings progress. Basically this should only return subzones that are in the plantings progress result. The code should be doing that by now. I haven't reviewed the merged response yet let me look :)

Reply:
Looks like my changes didn't make it in your merge - can you add them in?

Reply:
oops nevermind i see them now, sorry